pkgs/development/python-modules/freud/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, cython
+, oldest-supported-numpy
+, scikit-build
+, setuptools
+, wheel
+, tbb
+, numpy
+, rowan
+, scipy
+}:
+
+buildPythonPackage rec {
+  pname = "freud";
+  version = "3.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "glotzerlab";
+    repo = "freud";
+    rev = "v${version}";
+    hash = "sha256-aKh2Gub1vU/wzvWkCl8yzlIswp8CtR975USiCr6ijUI=";
+    fetchSubmodules = true;
+  };
+  # Because we prefer to not `leaveDotGit`, we need to fool upstream into
+  # thinking we left the .git files in the submodules, so cmake won't think we
+  # didn't initialize them.
+  preConfigure = builtins.map (submodule: ''
+    touch extern/${submodule}/.git
+  '') [ "fsph" "Eigen" "voro++" ];
+
+  nativeBuildInputs = [
+    cmake
+    cython
+    oldest-supported-numpy
```

Comment:
We should probably patch that tp just numpy

Reply:
> We should probably patch that tp just numpy

There is a difference between the two, and many packages require `oldest-supported-numpy`. TBH I don't understand the idea behind this meta package. Note that even in Nixpkgs it is defined to simply propagate numpy so I think it's OK.

Reply:
It is just a meta package which requires the last numpy upstream support IIRC